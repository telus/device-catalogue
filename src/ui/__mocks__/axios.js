import mockedAxios from "jest-mock-axios";

export default mockedAxios;