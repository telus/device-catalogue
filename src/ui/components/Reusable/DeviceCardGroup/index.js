import React from 'react';
import FlexGrid from '@tds/core-flex-grid';
import DeviceCard from '../DeviceCard';
const DeviceCardGroup = ({ data, img }) => {

  return (
      <FlexGrid id="device-card-group">
        <FlexGrid.Row>
          {data.map((item, index) => {
              const { brand, name, price } = item;
              return (
                  <FlexGrid.Col key={index}>
                    <DeviceCard brand={brand} name={name} price={price} img={img} />
                  </FlexGrid.Col>
              )
            })
          }
        </FlexGrid.Row>
      </FlexGrid>
  )
}

export default DeviceCardGroup;