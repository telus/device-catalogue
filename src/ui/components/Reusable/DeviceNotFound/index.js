import React from 'react';
import Heading from '@tds/core-heading';

const DeviceNotFound = () => {
  return (
    <div className="not-found">
      <Heading level="h1">
        Looks like the devices requested cannot be found.
      </Heading>
      <br/>
    </div>
  )
}

export default DeviceNotFound;
