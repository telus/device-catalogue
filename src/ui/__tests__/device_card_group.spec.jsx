import React from 'react';
import renderer from 'react-test-renderer';
import DeviceCardGroup from '../components/Reusable/DeviceCardGroup';
import DeviceNotFound from '../components/Reusable/DeviceNotFound';
import axios from 'axios';
const data = axios.get("http://localhost:8081/watches/").then(res => {
  return res.data.data;
});

describe('Device card group', () => {
  it('renders correctly', () => {
    const tree = renderer
      .create(<DeviceCardGroup data={data} img={''}/>)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders 404 if the data is corrupted', () => {
    const tree = renderer
      .create(<DeviceNotFound/>)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
