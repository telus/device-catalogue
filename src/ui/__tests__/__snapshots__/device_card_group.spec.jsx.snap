// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Device card group renders 404 if the data is corrupted 1`] = `
<div
  className="not-found"
>
  <h1
    className="Heading__StyledHeading-mpz92r-0 wprAR"
    data-testid="heading"
  >
    Looks like the devices requested cannot be found.
  </h1>
  <br />
</div>
`;
